src/ui/public/agg_types/filter/agg_type_filters.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { BehaviorSubject } from 'rxjs';
+import { AggType } from '..';
+import { AggConfig, Vis } from '../../vis';
+
+type AggTypeFilter = (
+  aggType: AggType,
+  vis: Vis,
+  aggConfig: AggConfig
+) => boolean;
+
+/**
+ * A registry to store {@link AggTypeFilter} which are used to filter down
+ * available aggregations for a specific visualization and {@link AggConfig}.
+ */
+class AggTypeFilters {
+  private filters = new Set<AggTypeFilter>();
+  private subject = new BehaviorSubject<Set<AggTypeFilter>>(this.filters);
+
+  /**
+   * Register a new {@link AggTypeFilter} with this registry.
+   * This will emit a new set of filtered aggTypes on every Observer returned
+   * by the {@link #filter$|filter method}.
+   *
+   * @param filter The filter to register.
+   */
+  public register(filter: AggTypeFilter): void {
```

Comment:
I would find this clearer to understand if the method was named `addFilter` -- `register` doesn't carry any implicit meaning for me, and it also raises some possible confusion with the registry system we currently use to communicate between plugins.